codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/IdempotencyTokenMiddlewareGenerator.java

```diff
@@ -151,12 +159,24 @@ private MemberShape getMemberWithIdempotencyToken(Model model, OperationShape op
                             }
                             return false;
                         })
-                        .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
-                            writer.write("$L.Initialize.Add(&$L{},middleware.After)",
-                                    stackOperand,
-                                    getIdempotencyTokenMiddlewareName(operation));
+                        .writeMiddlewareHelpers((writer, service, operation, protocolGenerator) -> {
+                            // Generate idempotency token middleware registrar function
                             writer.addUseImports(SmithyGoDependency.SMITHY_MIDDLEWARE);
+                            writer.write("// addIdempotencyTokenMiddleware registers middleware into operation stack");
+                            writer.openBlock("func addIdempotencyTokenMiddleware("
+                                            + "stack *middleware.Stack, cfg IdempotencyTokenProvider) {",
+                                    "}", () -> {
+                                writer.write("stack.Initialize.Add(&$L{cfg},middleware.After)",
+                                        getIdempotencyTokenMiddlewareName(operation));
+                            });
                         })
```

Comment:
Doesn't this `func` need to be specific to the individual operation with the middleware? It looks like a single `addIdempotencyTokenMiddleware` func will be defined that will add a operation specific middleware.

Reply:
The middleware's are operation specific. So the middleware helpers have to be operation specific. What do you suggest otherwise?